### PR TITLE
[Online-DPO] num_generation_per_prompt is fixed

### PR DESCRIPTION
**Pull request by kashif**
* `num_generation_per_prompt` is always 2 for the online DPO case
* removed un-needed `with torch.no_grad():`

### CODE REVIEW

**Comment by qgallouedec** [comment]
Important: seems like eos penalty had an error

```diff
- scores = torch.where(contain_eos_token, scores, args.penalty_reward_value)
+ scores = torch.where(contain_eos_token, args.penalty_reward_value, scores)
```

File: trl/trainer/online_dpo_trainer.py [hunk 1]
```diff
@@ -287,14 +288,13 @@ def repeat_generator():
                     del logits, all_logprob
                     torch.cuda.empty_cache()
 
-                    with torch.no_grad():
```
Comment (qgallouedec) [inline trl/trainer/online_dpo_trainer.py:290]: it's because ref model is in eval mode that you don't need to use the context?

**Review by qgallouedec — APPROVED** [review]
